File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30Strategy/src/main/java/ec/edu/espe/controller/QuickSort.java

package ec.edu.espe.controller;

/**
 *
 * @author MSI-PULSE
 */
public class QuickSort implements SortingStrategy {

    @Override
    public void sort(int[] numbers) {
        quickSort(numbers, 0, numbers.length - 1);
    }

    private void quickSort(int[] numbers, int low, int high) {
        if (low < high) {
            int pi = partition(numbers, low, high);
            quickSort(numbers, low, pi - 1);
            quickSort(numbers, pi + 1, high);
        }
    }

    private int partition(int[] numbers, int low, int high) {
        int pivot = numbers[high];
        int i = (low - 1);
        for (int j = low; j < high; j++) {
            if (numbers[j] < pivot) {
                i++;
                int temp = numbers[i];
                numbers[i] = numbers[j];
                numbers[j] = temp;
            }
        }
        int temp = numbers[i + 1];
        numbers[i + 1] = numbers[high];
        numbers[high] = temp;

        return i + 1;
    }

    @Override
    public String getStrategyName() {
        return "Quick Sort";
    }
}

File: exams/viteri/U3/FinalExam Viteri/FinalExam Viteri/utils/Quick_Sort.py
class Quick_Sort:
    def quick_sort(list):
        for (int i = 0; i < n-1; i++) 
            for (int j = 0; j < n-i-1; j++) 
                if (list[j] > list[j+1]) 
                    int temp = list[j];
                    list[j] = list[j+1];
                    list[j+1] = temp;
                